src/main/java/com/glencoesoftware/bioformats2raw/Converter.java

```diff
@@ -732,13 +714,45 @@ private void processTile(
     }
 
     Slf4JStopWatch t1 = stopWatch();
+    ByteBuffer tileAsByteBuffer = ByteBuffer.wrap(tileAsBytes);
     try {
-      n5.writeBlock(
-        pathName,
-        n5.getDatasetAttributes(pathName),
-        dataBlock
-      );
-      LOGGER.info("successfully wrote at {} to {}", gridPosition, pathName);
+      switch (pixelType) {
```

Comment:
If we could envision a way to not need multiple versions of this `switch`-block, that would likely be a nice utility to have at some lower layer.

Reply:
I don't think there's any way of avoiding having at least two. They are now available as static methods as of fb822fe2557e06b98739e6645dbfc785b6afbeeb.